feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch

```diff
@@ -0,0 +1,1489 @@
++        target_event_record_t  *target_record = NULL;
++        ds_dlist_iter_t         record_iter;
++        /* get cached eventes */
++        for (target_record = ds_dlist_ifirst(&record_iter, &g_rt_events.event_list);
++             target_record != NULL;
++             target_record = ds_dlist_inext(&record_iter))
++        {
++            // copy (shallow)
++	    ds_dlist_insert_tail(nl_call_param->list, target_record);
++            // remove
++            ds_dlist_iremove(&record_iter);
++        }
++        // no error
++        return 0;
++}
++
+ int stats_nl80211_init(void)
+ {
+ 	if (unl_genl_init(&unl, "nl80211") < 0) {
+Index: opensync-2.0.5.0/src/lib/datapipeline/inc/dppline.h
+===================================================================
+--- opensync-2.0.5.0.orig/src/lib/datapipeline/inc/dppline.h
++++ opensync-2.0.5.0/src/lib/datapipeline/inc/dppline.h
+@@ -46,6 +46,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBI
+ #include "dpp_bs_client.h"
+ #include "dpp_rssi.h"
+ #include "dpp_network_probe.h"
++#include "dpp_events.h"
+ 
+ #ifdef CONFIG_MANAGER_QM
+ // QM does queue-ing of reports when offline on it's own, so dpp needs
+@@ -107,6 +108,11 @@ bool dpp_put_bs_client(dpp_bs_client_rep
+ bool dpp_put_rssi(dpp_rssi_report_data_t *rpt);
+ 
+ /*
++ * Put event report to internal queue
++ */
++bool dpp_put_events(dpp_event_report_data_t *rpt);
++
++/*
+  * Get the protobuf packed buffer
+  *
+  * This buffer is ready to be send using MQTT
+Index: opensync-2.0.5.0/src/sm/src/sm_ovsdb.c
+===================================================================
+--- opensync-2.0.5.0.orig/src/sm/src/sm_ovsdb.c
++++ opensync-2.0.5.0/src/sm/src/sm_ovsdb.c
+@@ -62,6 +62,7 @@ char *sm_report_type_str[STS_REPORT_MAX]
+     "device",
+     "rssi",
+     "network_probe",
++    "event",
+ };
+ 
+ #ifndef CONFIG_MANAGER_QM
+@@ -359,8 +360,11 @@ bool sm_update_stats_config(sm_stats_con
+             sm_rssi_report_request(&radio->config, &req);
+             break;
+ 	case STS_REPORT_NETWORK_PROBE:
+-	sm_network_probe_report_request(&req);
+-	break;
++	    sm_network_probe_report_request(&req);
++	    break;
++        case STS_REPORT_EVENT:
++            sm_events_report_request(&radio->config, &req);
++            break;
+         default:
+             return false;
+     }
+Index: opensync-2.0.5.0/src/lib/target/inc/target_native.h
+===================================================================
+--- opensync-2.0.5.0.orig/src/lib/target/inc/target_native.h
++++ opensync-2.0.5.0/src/lib/target/inc/target_native.h
+@@ -39,6 +39,8 @@ typedef void target_survey_record_t;
+ 
+ typedef void target_capacity_data_t;
+ 
++typedef void target_event_record_t;
++
+ #include "target_common.h"
+ 
+ #endif  /* TARGET_NATIVE_H_INCLUDED */
+Index: opensync-2.0.5.0/vendor/tip/src/lib/target/inc/target_tip.h
+===================================================================
+--- opensync-2.0.5.0.orig/vendor/tip/src/lib/target/inc/target_tip.h
```

Comment:
dont patches files in the src/ folder but apply the patch to the actual file please